pageserver/src/layered_repository.rs

```diff
@@ -195,12 +201,11 @@ impl Repository for LayeredRepository {
         let _gc_cs = self.gc_cs.lock().unwrap();
 
         let mut timelines = self.timelines.lock().unwrap();
-        let src_timeline = match self.get_or_init_timeline(src, &mut timelines)? {
-            LayeredTimelineEntry::Local(timeline) => timeline,
-            LayeredTimelineEntry::Remote { .. } => {
-                bail!("Cannot branch off the timeline {} that's not local", src)
-            }
-        };
+        let src_timeline = self
+            .get_local_timeline_load_internal(src, &mut timelines)
+            // message about timeline being remote is one .context up in the stack
+            .context("failed to load timeline for branching")?
+            .ok_or(anyhow::anyhow!("unknown timeline id: {}", &src))?;
```

Comment:
NIT: might be more "right" to use `.ok_or_else(|| ...)` to defer the formatting, there are a few more placed like that. Also worth importing `anyhow::anyhow` at this point too, I guess.